src/main/python/linuxband/midi/midi_player.py

```diff
@@ -105,15 +110,15 @@ def shutdown(self):
         # descriptors could have been already closed
         try:
             os.close(self.__piper)
-        except:
+        except IOError:
```

Comment:
I think os.close() can throw OSError as well. To remain on the safe side I would better stick with catching all possible exceptions.
Btw. swallowing the exception like this is a bad code. It would have been much better to log the exception.